### PR TITLE
WebKit export of https://bugs.webkit.org/show_bug.cgi?id=188189

### PR DESCRIPTION
**Pull request by darsh7807**
[WebKit export from bug: Perform a microtask checkpoint before creating a custom element](https://bugs.webkit.org/show_bug.cgi?id=188189)

### CODE REVIEW

**Comment by darsh7807** [comment]
This PR is same as https://github.com/web-platform-tests/wpt/pull/14221,
I messed up git history in previous PR so created new one

File: custom-elements/perform-microtask-checkpoint-before-construction.html [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+}, 'HTML parser must perform a microtask checkpoint before constructing a custom element');
+
+promise_test(async function () {
+    const contentWindow = await construct_custom_element_in_parser(this, '<b><i>hello</b><some-element>');
+    const contentDocument = contentWindow.document;
+    let recordsList = contentWindow.recordsListInConstructor;
+    assert_true(Array.isArray(recordsList));
+    assert_equals(recordsList.length, 1);
+    assert_true(Array.isArray(recordsList[0]));
+    console.log(recordsList[0]);
```
[the remark this answers is not in the document]
Comment (darsh7807) [reply custom-elements/perform-microtask-checkpoint-before-construction.html]: I added it for debugging of the test, not needed now, so removing it  now

File: custom-elements/perform-microtask-checkpoint-before-construction.html [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    assert_equals(record.nextSibling, null);
+    assert_equals(record.removedNodes.length, 0);
+    assert_equals(record.addedNodes.length, 1);
+    assert_equals(record.addedNodes[0], contentDocument.querySelector('some-element'));
+}, 'HTML parser must perform a microtask checkpoint before constructing a custom element during the adoption agency algorithm');
+
+</script>
+</body>
+</html>
+
```
Comment (fred-wang) [inline custom-elements/perform-microtask-checkpoint-before-construction.html]: can you please remove this extra line at the end?

Comment (darsh7807) [reply custom-elements/perform-microtask-checkpoint-before-construction.html]: Okay, I removed t

**Review by fred-wang — APPROVED** [review]
Thanks.

Can you please open a bug on https://bugs.chromium.org/p/chromium for the remaining failing test on chromium and cc' the relevant people.
https://github.com/web-platform-tests/wpt/pull/14221#issuecomment-441709119
https://github.com/web-platform-tests/wpt/pull/14221#issuecomment-441874584

**Comment by darsh7807** [comment]
@fred-wang  Created issue https://bugs.chromium.org/p/chromium/issues/detail?id=91082,
I am not able to find the option to cc people.

**Comment by rniwa** [comment]
That can't be right. The issue has been closed in 2012.

**Comment by darsh7807** [comment]
> @fred-wang Created issue https://bugs.chromium.org/p/chromium/issues/detail?id=91082,
> I am not able to find the option to cc people.

Sorry I missed the 3 at the end of link, correct link is 
https://bugs.chromium.org/p/chromium/issues/detail?id=910823

**Pull request merged**